pkg/cloudprovider/providers/aws/aws_test.go

```diff
@@ -266,12 +266,37 @@ func contains(haystack []*string, needle string) bool {
 
 func instanceMatchesFilter(instance *ec2.Instance, filter *ec2.Filter) bool {
 	name := *filter.Name
+	glog.Infof("Checking filter %v on instance id %v", name, *instance.InstanceId)
```

Comment:
Use aws.StringValue instead of dereferencing the possibly-nil InstanceId (admittedly unlikely in this case!)